rfcs/20180827-api-names.md

```diff
@@ -0,0 +1,587 @@
+| tf.sets.set_difference | tf.manip.set_difference |
+| tf.sets.set_intersection | tf.manip.set_intersection |
+| tf.sets.set_size | tf.manip.set_size |
+| tf.sets.set_union | tf.manip.set_union |
+| tf.sign | tf.math.sign |
+| tf.slice | tf.manip.slice |
+| tf.space_to_batch | tf.nn.space_to_batch, tf.manip.space_to_batch |
+| tf.space_to_depth | tf.nn.space_to_depth, tf.manip.space_to_depth |
+| tf.sparse_add | tf.sparse.add |
+| tf.sparse_concat | tf.sparse.concat |
+| tf.sparse_fill_empty_rows | tf.sparse.fill_empty_rows |
+| tf.sparse_mask | tf.sparse.mask |
+| tf.sparse_maximum | tf.sparse.maximum |
+| tf.sparse_merge | tf.sparse.merge |
+| tf.sparse_minimum | tf.sparse.minimum |
+| tf.sparse_placeholder | tf.sparse.placeholder |
+| tf.sparse_reduce_max | tf.sparse.reduce_max |
+| tf.sparse_reduce_max_sparse | tf.sparse.reduce_max_sparse |
+| tf.sparse_reduce_sum | tf.sparse.reduce_sum |
+| tf.sparse_reduce_sum_sparse | tf.sparse.reduce_sum_sparse |
+| tf.sparse_reorder | tf.sparse.reorder |
+| tf.sparse_reset_shape | tf.sparse.reset_shape |
+| tf.sparse_reshape | tf.sparse.reshape |
+| tf.sparse_retain | tf.sparse.retain |
+| tf.sparse_segment_mean | tf.sparse.segment_mean |
+| tf.sparse_segment_sqrt_n | tf.sparse.segment_sqrt_n |
+| tf.sparse_segment_sum | tf.sparse.segment_sum |
+| tf.sparse_slice | tf.sparse.slice |
+| tf.sparse_softmax | tf.sparse.softmax |
+| tf.sparse_split | tf.sparse.split |
+| tf.sparse_tensor_dense_matmul | tf.sparse.matmul |
+| tf.sparse_tensor_to_dense | tf.sparse.sparse_tensor_to_dense |
+| tf.sparse_to_dense | tf.sparse.to_dense |
+| tf.sparse_to_indicator | tf.sparse.to_indicator |
+| tf.sparse_transpose | tf.sparse.transpose |
+| tf.split | tf.manip.split |
+| tf.sqrt | tf.math.sqrt |
+| tf.square | tf.math.square |
+| tf.squeeze | tf.manip.squeeze |
+| tf.stack | tf.manip.stack |
+| tf.strided_slice | tf.manip.strided_slice |
+| tf.string | tf.dtypes.string |
+| tf.string_split | tf.strings.split |
+| tf.subtract | tf.math.subtract |
+| tf.tables_initializer | tf.initializers.tables_initializer |
+| tf.tanh | tf.math.tanh, tf.nn.tanh |
+| tf.train.match_filenames_once | tf.io.match_filenames_once |
+| tf.train.write_graph | tf.io.write_graph |
+| tf.transpose | tf.manip.transpose |
+| tf.truediv | tf.math.truediv |
+| tf.truncated_normal | tf.random.truncated_normal |
+| tf.truncatediv | tf.math.truncatediv |
+| tf.truncatemod | tf.math.truncatemod |
+| tf.uint16 | tf.dtypes.uint16 |
+| tf.uint32 | tf.dtypes.uint32 |
+| tf.uint64 | tf.dtypes.uint64 |
+| tf.uint8 | tf.dtypes.uint8 |
+| tf.unique | tf.manip.unique |
+| tf.unique_with_counts | tf.manip.unique_with_counts |
+| tf.unsorted_segment_mean | tf.math.unsorted_segment_mean |
+| tf.unsorted_segment_sqrt_n | tf.math.unsorted_segment_sqrt_n |
+| tf.unstack | tf.manip.unstack |
+| tf.variant | tf.dtypes.variant |
+| tf.verify_tensor_all_finite | tf.debugging.verify_tensor_all_finite |
+
+
+
+
+
+# Appendix 2: Deprecated Endpoints
```

Comment:
Should we deprecate the tf.logging module? 

We clearly need some facility that TF can use internally, but I'm not sure that other people need to use the same? Maybe we keep a single endpoint with which users can get the logger, and then use that using their own import of python's logging?

Just a suggestion, triggered by https://github.com/tensorflow/tensorflow/issues/21930, which made clear to me that we're just providing a wrapper around logging without adding any real value.

Reply:
Just talked about this with @tensorflow/api-owners and we do want to remove tf.logging from the public API (we will continue to use it for TF itself, but it's unclear what value, if any, it provides to users over just using Python's logging module.

Reply:
I added a note about deprecating tf.logging under "Deprecated namespaces"